### PR TITLE
Vmo 2596 documentation

### PR DESCRIPTION
**Pull request by jacob-viamo**
Initial documentation for the purpose, usage and setup of the flow runner based on the checklist in https://viamoinc.atlassian.net/browse/VMO-2596 

A couple of things are missing though. First, the /trees/create route is not working for me. Am I missing some bit of setup here or is this still TODO?

Second, I didn't feel I could give a good write up on this item:

> General explanation of how flow builder is coupled to flow runner (this may require @JacobAshdown and @BrettZabos to connect briefly)

Beyond saying that it can author flows which can then be run in flow runner. @bzabos I'd appreciate your input here once you're back!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,64 @@
...
+## What are flows
+
+Flows are a modern paradign for describing the logic of digital information systems that interact with individuals,
+often for the purpose of (a) collecting data or (b) providing information through interactive requests. Some common examples of this are in mobile services using voice-based or SMS-based conversations over basic mobile phones. Flows follow the "flowchart" paradigm, consisting of actions (nodes) and connections between actions, which can incorporate decision-making logic.
+
+## Usage
+
+To begin, you can create a new flow by visiting [localhost:8080/trees/create](http://localhost:8080/trees/create). From there, you can either start with a blank flow (a flow without blocks) and get editing:
+
+![Flow without blocks](./docs/images/screenshot-no-blocks.png)
```
[the remark this answers is not in the document]
Comment (jacob-viamo) [reply README.md:14]: Hmm, my intention was to show what the json would look like for such a flow but maybe your suggestion is clearer.

File: README.md [hunk 2]
```diff
@@ -1,29 +1,64 @@
 # flow-builder
 
+flow-builder provides an open source frontend for authoring [flow interoperability project specification](https://floip.gitbook.io/flow-specification/) compliant flows. It enables the importing, editing and exporting of flow json, as well as authoring from scratch. Such flows can then be run using a compatible flow runner such as [FLOIP/flow-runner](https://github.com/FLOIP/flow-runner).
+
+## What are flows
```
[the remark this answers is not in the document]
Comment (jacob-viamo) [reply README.md:5]: I'd prefer to leave as is. I think someone skimming could easily miss this important context.

File: README.md [hunk 3]
```diff
@@ -1,29 +1,64 @@
...
+
+Flows are a modern paradign for describing the logic of digital information systems that interact with individuals,
+often for the purpose of (a) collecting data or (b) providing information through interactive requests. Some common examples of this are in mobile services using voice-based or SMS-based conversations over basic mobile phones. Flows follow the "flowchart" paradigm, consisting of actions (nodes) and connections between actions, which can incorporate decision-making logic.
+
+## Usage
```
[the remark this answers is not in the document]
Comment (jacob-viamo) [reply README.md]: Done

File: README.md [hunk 4]
```diff
@@ -1,29 +1,64 @@
...
-```
-
 ### Lints and fixes files
 ```
 yarn lint
 ```
 
-### Customize configuration
-See [Configuration Reference](https://cli.vuejs.org/config/).
+## An explanation of the app.config.json file vs the builder.config.json file
```
[the remark this answers is not in the document]
Comment (jacob-viamo) [reply README.md]: Done

File: README.md [hunk 5]
```diff
@@ -1,29 +1,64 @@
...
+Flows are a modern paradign for describing the logic of digital information systems that interact with individuals,
+often for the purpose of (a) collecting data or (b) providing information through interactive requests. Some common examples of this are in mobile services using voice-based or SMS-based conversations over basic mobile phones. Flows follow the "flowchart" paradigm, consisting of actions (nodes) and connections between actions, which can incorporate decision-making logic.
+
+## Usage
+
+To begin, you can create a new flow by visiting [localhost:8080/trees/create](http://localhost:8080/trees/create). From there, you can either start with a blank flow (a flow without blocks) and get editing:
+
+![Flow without blocks](./docs/images/screenshot-no-blocks.png)
+
+Or you can use the Import/Export panel and text field to copy in an existing flow:
```
[the remark this answers is not in the document]
Comment (jacob-viamo) [reply README.md]: Done

**Review by bzabos — APPROVED** [review]
Nice, thanks! I appreciate the list of _Future improvements_ at the bottom. 

@jacob-viamo do we have this list elsewhere or is this document the only place it's been compiled?

**Comment by bzabos** [comment]
@jacob-viamo @safydy-r I have not done anything with `/trees/create` — I will likely just provide a single URL, remove the example from `master` and we can provide a branch with some sample JSON + configuration if we so choose. Thoughts?

**Comment by jcbashdown** [comment]
>@jacob-viamo do we have this list elsewhere or is this document the only place it's been compiled?

Nothing yet. Most of these need to happen when we actually add the feature - e.g. validations, integration back to voto5 etc. I think there are stories for most/all of these.

>@jacob-viamo @safydy-r I have not done anything with /trees/create — I will likely just provide a single URL, remove the example from master and we can provide a branch with some sample JSON + configuration if we so choose. Thoughts?

Not too sure what you mean here. I basically think this just needs to document the url which will be used to create a fresh flow. Do you know what that will be?

**Pull request merged**